src/lib/spinel/radio_spinel_impl.hpp

```diff
@@ -1901,6 +1908,24 @@ otError RadioSpinel<InterfaceType, ProcessContextType>::Transmit(otRadioFrame &a
 
     mTransmitFrame = &aFrame;
 
+#if OPENTHREAD_CONFIG_TIME_SYNC_ENABLE
+    if (mTransmitFrame->mInfo.mTxInfo.mIeInfo->mTimeIeOffset != 0)
+    {
+        uint8_t *timeIe = mTransmitFrame->mPsdu + mTransmitFrame->mInfo.mTxInfo.mIeInfo->mTimeIeOffset;
+        uint64_t time   = otPlatTimeGet() + mTransmitFrame->mInfo.mTxInfo.mIeInfo->mNetworkTimeOffset;
```

Comment:
I think it's worth to leave the casts from the original code to silence compiler warnings.